platform/ui/src/components/StepProgressDropdown/StepProgressDropdown.tsx

```diff
@@ -0,0 +1,326 @@
+          'h-1 grow mr-1 last:mr-0 first:rounded-l-sm last:rounded-r-sm',
+          {
+            'bg-black': !activated && !completed,
+            'bg-primary-main': activated && !completed,
+            'bg-primary-light': completed,
+          }
+        )}
+      ></div>
+    );
+  }
+
+  return <div className="flex">{items}</div>;
+};
+
+StepProgressStatus.propTypes = {
+  options: PropTypes.arrayOf(DROPDOWN_OPTION_PROPTYPE).isRequired,
+};
+
+const StepProgressDropdownItemContent = ({ option, debugEnabled }) => {
+  const { label, info, completed } = option;
+  const [truncate, setTruncate] = useState(true);
+  const handleOnHideTooltip = () => setTruncate(true);
+  let icon;
+
+  if (completed) {
+    icon = 'status-tracked';
+  } else if (info) {
+    icon = 'launch-info';
+  }
+
+  const tooltipText = useMemo(() => {
+    if (!truncate || !info || info.length <= MAX_TOOLTIP_LENGTH) {
+      return info;
+    }
+
+    const handleReadMoreClick = e => {
+      setTruncate(false);
+      e.stopPropagation();
+      e.preventDefault();
+    };
+
+    return (
+      <>
+        {info.substr(0, MAX_TOOLTIP_LENGTH)}
+        <button
+          className="text-primary-active font-bold"
+          onClick={handleReadMoreClick}
+        >
+          &nbsp;Read more...
+        </button>
+      </>
+    );
+  }, [info, truncate]);
+
+  const iconClassNames =
+    'grow-0 text-primary-light h-4 w-4 mt-1 mr-2 mb-0 ml-1';
+
+  const iconContent = (
+    <>
+      {icon && <Icon name={icon} className={iconClassNames} />}
+      {!icon && <div className={iconClassNames} />}
+    </>
+  );
+
+  return (
+    <>
+      {info && (
+        <Tooltip
+          content={tooltipText}
+          position="bottom-left"
+          tooltipBoxClassName={'max-w-xs'}
+          onHide={handleOnHideTooltip}
+        >
+          {iconContent}
+        </Tooltip>
+      )}
+      {!info && iconContent}
+
+      <div className="grow text-base leading-6">{label}</div>
+    </>
+  );
+};
+
+StepProgressDropdownItemContent.propTypes = {
+  option: DROPDOWN_OPTION_PROPTYPE.isRequired,
+};
+
+const StepProgressDropdownItem = ({ option, onSelect }) => {
+  const { value } = option;
+
+  return (
+    <div
+      key={value}
+      className={
+        'flex py-1 cursor-pointer hover:bg-secondary-main transition duration-1000'
+      }
+      onClick={() => onSelect(option)}
+    >
+      <StepProgressDropdownItemContent option={option} />
+    </div>
+  );
+};
+
+StepProgressDropdownItem.propTypes = {
+  option: DROPDOWN_OPTION_PROPTYPE.isRequired,
+  onSelect: PropTypes.func,
+};
+
+const StepProgressDropdown = ({
+  options: optionsProps,
+  value,
+  children,
+  onChange,
+}: {
+  options: StepProgressDropdownOption[];
+  value?: string;
+  children?: ReactNode;
+  onChange?: ({ selectedOption }) => void;
+}): JSX.Element => {
+  const element = useRef(null);
+  const [open, setOpen] = useState(false);
+  const toggleOptions = () => setOpen(s => !s);
```

Comment:
why this is called `toggleOptions` and what it does is toggling open? should it be `toggleOpen`?

Reply:
It's called `toggleList` in `Dropdown.tsx` and I had renamed everything that contains `list` to `options`. It's now renamed to `toggleOpen`.